pkg/compilers/osv/generic.go

```diff
@@ -4,6 +4,35 @@ import (
 	"github.com/emc-advanced-dev/unik/pkg/types"
 )
 
+// OSvCompilerBase should be embedded in all OSv compilers.
 type OSvCompilerBase struct {
-	CreateImage func(params types.CompileImageParams, useEc2Bootstrap bool) (string, error)
+	CompilerHelper CompilerHelper
+}
+
+// ConvertParams contains all the information needed when bootstrapping.
+type ConvertParams struct {
+	// CapstanImagePath points to image that was composed by Capstan
+	CapstanImagePath string
+
+	// CompileParams stores parameters that were used for composing image
+	CompileParams types.CompileImageParams
+}
+
+// CompilerHelper implements conversion of Capstan result into provider-like image.
+// It should be implemented per provider. In this context converting means e.g.
+// converting .qcow2 image (Capstan result) into .wmdk for VirtualBox provider.
+type CompilerHelper interface {
+	// Convert converts Capstan-provided image into appropriate format
+	// for the provider
+	Convert(params ConvertParams) (*types.RawImage, error)
+
+	// UseEc2 tells whether or not to prepare image for EC2 (Elastic Compute Cloud)
+	UseEc2() bool
+}
+
+// CompilerHelperBase should be embedded in every CompilerHelper implementation.
```

Comment:
I'd say that everyone should embed it and then override `UseEc2()` if needed. Isn't it more readable this way?